### PR TITLE
Yongjian - Fix minor alignment of assign team button on teams tab

### PR DESCRIPTION
**Pull request by ypan710**
# Description
Screen Shot 2023-06-01 at 8 00 31 PM

## Mainly changes explained:
Adjusted width of "Assign Team" button under "Teams Tab" under user profile page.

## How to test:
1. check into current branch
2. do `npm run start:local` to run this PR locally
3. login as admin role -> user profile under dashboard -> Teams Tab -> inspect the change in width of the "Assign Team" button.

## Screenshots
Screen Shot 2023-06-01 at 8 01 42 PM

### CODE REVIEW

**Comment by eduhorta** [comment]
Hi, @ypan710!

How about placing the button on the right corner?
image

**Review by nathanah — CHANGES_REQUESTED** [review]
Just changing the width here won't make it reactive. As the other reviewers said, it should be right aligned with the title filling the rest of the space.

**Comment by yufeiStella** [comment]
Below is the modified screenshot. It partially worked, but it's still not aligned properly. How about moving the blue button labeled "Assign Team" to the right and extending the gray background on the left side of the "Teams"? 
Screen Shot 2023-06-02 at 23 19 37

**Review by NidaZaki — CHANGES_REQUESTED** [review]
I agree with other reviewers on maybe placing the button to the right corner.
PR#867

Also, the alignment for this button got changed:
PR#867-1

**Review by OAkinbode — CHANGES_REQUESTED** [review]
Hello Yongjian, I agree with the rest of the reviewers that the button needs to go to the right of that component. Something like below: 
PR 867 changes needed

**Review by papiasharmin — CHANGES_REQUESTED** [review]
Hello @ypan710 ,only  this width change  will not do, I think it will need some adjustment with previous style to get the desired result.

File: src/components/UserProfile/TeamsAndProjects/UserTeamsTable.jsx
Comment (nathanah) [inline src/components/UserProfile/TeamsAndProjects/UserTeamsTable.jsx:1]: Your changes still have an overhang for 1025-1199px. This is why I replaced the \<Col\> elements in my suggested changes. The table below does not take up the same width as md=12.
image

File: src/components/UserProfile/TeamsAndProjects/UserProjectsTable.jsx
Comment (nathanah) [inline src/components/UserProfile/TeamsAndProjects/UserProjectsTable.jsx:1]: Same issue as TeamsTable.

**Review by nathanah — CHANGES_REQUESTED** [review]

**Review by cvtqx — CHANGES_REQUESTED** [review]
Hi @ypan710 ! 

The button looks good on large screen, bu it overhangs the table on a medium screen (below 1100px). See pictures attached:

Large screen:

Screen Shot 2023-06-07 at 2 19 49 PM

Medium screen:

Screen Shot 2023-06-07 at 2 27 56 PM

**Review by papiasharmin — CHANGES_REQUESTED** [review]
Hello, @ypan710 , I think the grid should be 9:3 instead of 10:2, TeamsAndProjects.css  for .btn-addteam width should be 100% and remove margin-left. Better will be if you just use flex-grow property instead of grid, its easy and clean.

**Comment by eduhorta** [comment]
Hi, @ypan710!

On large screens it is working as intended:
image

But on 1200px breakpoint:
image

**Review by NidaZaki — COMMENTED** [review]
The "Assign Team" button looks good, but the "Assign Project" button seems off.

image
image

File: src/components/UserProfile/TeamsAndProjects/TeamsAndProjects.css [hunk 1]
```diff
@@ -3,13 +3,36 @@
...
+  padding: 0px;
+}
+
+.teamtable-container .row,
+.projecttable-container .row {
+  margin: 0px;
+}
+
 .btn-addteam,
 .btn-addproject {
```
Comment (nathanah) [inline src/components/UserProfile/TeamsAndProjects/TeamsAndProjects.css:31]:
```suggestion
.btn-addproject {
  width: 100%;
```
Before:
image
After:
image

File: src/components/UserProfile/TeamsAndProjects/UserTeamsTable.jsx [hunk 2]
```diff
@@ -28,18 +28,27 @@ const UserTeamsTable = props => {
...
+            </Col> */}
+            <div className="col"
+              style={{
+                backgroundColor: ' #e9ecef',
+                border: '1px solid #ced4da',
+                marginBottom: '10px',
+              }}
+            >
+              <span className="teams-span">Teams</span>
+            </div>
```
Comment (nathanah) [inline src/components/UserProfile/TeamsAndProjects/UserTeamsTable.jsx:49]:
```suggestion
            <Col
              md={props.edit ? '10' : '12'}
              style={{
                backgroundColor: ' #e9ecef',
                border: '1px solid #ced4da',
                marginBottom: '10px',
              }}
            >
              <span className="teams-span">Teams</span>
            </Col>
```
Should stay consistent with the rest.

**Review by nathanah — CHANGES_REQUESTED** [review]

**Comment by rachnapandya** [comment]
It is fixed.
teamstab_align
I would suggest aligning the visibility tab to the left if possible.
teamstab_align

File: src/components/UserProfile/TeamsAndProjects/UserTeamsTable.jsx (hunk 2 above)
Comment (ypan710) [reply src/components/UserProfile/TeamsAndProjects/UserTeamsTable.jsx]: Please re-review my newest commit. Thanks.

**Review by bienzguoa — DISMISSED** [review]
It looks good. It might be off-topic, but I would also suggest making the visible toggle to be aligned in the right, and adjust "assign project" like " assign team".
Screenshot 2023-06-17 at 1 56 36 AM
Screenshot 2023-06-17 at 1 57 26 AM

**Comment by one-community** [comment]
> It looks good. It might be off-topic, but I would also suggest making the visible toggle to be aligned in the right, and adjust "assign project" like " assign team".

Yes, please make these changes too.

**Review by KurtisIvey — APPROVED** [review]
Reviewed and it seems like the overhang from 1025-1199px has been addressed. I noticed that people were suggesting changes to the visibility row, so I went ahead and tried to address that myself in my own code snippet. The solution was rather simple since my approach remained fully responsive and ditched the annoying bootstrap <Col>. In addition, teams tab uses a modal on smaller res. There seemed to be way too many div containers, so I deleted them which allowed me to implement an easy fix.

Approving because visibility row isn't necessarily in the scope

alignment is good:
pr867 works


why so many divvvssss :(    
pr 867 suggestion


visibility tab realignment solution(Good old fashion flexbox and padding):
pr867 visibility row solution

**Comment by beblicarl** [comment]
This works as intended however I think it would be cool if the visible-invisible switch were vertically aligned with the "Assign Team" button

Assign-teams-button 2023-06-29 122821

**Comment by one-community** [comment]
> visibility tab realignment solution(Good old fashion flexbox and padding):

Yes, please update to include this

**Review by mahamakhan — DISMISSED** [review]
Looks great.

image

**Review by yaow62023 — DISMISSED** [review]
Assign team button width adjusted. Great work! 

Screenshot 2023-07-01 at 11 08 13 AM

**Comment by Jiangwei-shi** [comment]
I've adjusted various widths of the browser and everything works well now. The last image shows the smallest adjustment, where clicking on 'team' will open a new window. I'm not sure if this design meets the requirements.
image
image
image

**Comment by Shreyj14** [comment]
Looking at the previous reviews the changes look good. One theing which can be done is that having button inside the Team div itself, so it wont have that separate boundary.
image

**Review by Shreyj14 — APPROVED** [review]

**Review by HectorAgudelo — COMMENTED** [review]
There is still some issues when the screen size is between 1025-1199 px. the whole row height seems to be off.
PR#867-2
PR#867

**Review by eduvarjaor — DISMISSED** [review]
I tested this PR and looks good. Good job!

Screenshot_1

**Review by yurisokolovicz — DISMISSED** [review]
Hi @ypan710,

I approve this PR. The button has now been properly corrected with shorter dimensions.
Before:
Screenshot from 2023-07-26 15-40-34
After:
Screenshot from 2023-07-26 15-41-47

**Review by Indujaks — APPROVED** [review]
Tested the changes, it works as expected!
Approved!
Screenshot 2023-07-27 at 3 47 55 PM

As suggested in the above reviews, the same thing can be implemented for Assign project as well
Screenshot 2023-07-27 at 3 48 18 PM

**Comment by Lin1404** [comment]
Hi @ypan710,

I tested this PR and change different width of my browser. It looks awesome. Approved!

image
image
image

**Review by Lin1404 — DISMISSED** [review]

**Review by zuhang2000 — APPROVED** [review]
Hi I test your PR and it works as expected! I also test the behavior if I zoom in or out and it seems good to me. Good job!

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/123051008/7c1c9c21-dd60-4e72-a3e3-b3379ff184a4

**Review by ThapeloMasasa — APPROVED** [review]
The alignment of the assign button is great and it is also responsive. Good job
Screenshot 2023-08-02 at 9 11 32 PM

**Review by EdwinLauM — APPROVED** [review]
Hey @ypan710! Tested with different widths and works as intended!

Screen Shot 2023-08-02 at 11 35 14 PM

Screen Shot 2023-08-02 at 11 35 56 PM

**Review by DavidC0126 — COMMENTED** [review]
Hi I test your PR and it works as described. I also test the behavior if I zoom in or out and it seems good to me. 
team1
team2

**Review by Abiddy — DISMISSED** [review]
LGTM!
Screenshot from 2023-08-05 23-29-51

**Comment by iTvX** [comment]
As I checked on my side, the assign Team button looks perfect.  Great job!

**Review by harshidad — APPROVED** [review]
Buttton aligned as mentioned. 

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/109878936/ea370eb7-c06e-44c9-a875-b70574671df3

**Review by Chuehleo — APPROVED** [review]
Everything works as intended, great job!
Screenshot 2023-08-18 134542

**Review by ss23046 — CHANGES_REQUESTED** [review]
Hi @ypan710 , I tested the PR
Large Screen : Both look good!

Screen Shot 2023-08-19 at 1 27 05 AM
Screen Shot 2023-08-19 at 1 27 20 AM

Medium Screen: 'Assign team' is fine, but 'Assign project' needs to be changed a bit(Perhaps, make it look full width like 'Assign team', else, adjust it to the left  
Screen Shot 2023-08-19 at 1 28 13 AM

**Comment by AnishPandita** [comment]
Hey, the results look as intended. Tested on different sizes.
Screenshot 2023-08-25 at 3 52 36 PM

**Comment by Shubhu0** [comment]
Checked PR, all sizes look good

**Comment by one-community** [comment]
> Hey, the results look as intended. Tested on different sizes. Screenshot 2023-08-25 at 3 52 36 PM

In this picture example, the delete button should be centered. Please fix.

**Comment by one-community** [comment]
Please confirm this issue is fixed with this PR too: 
image
This happens when a team exists and you click "delete".

**Review by cvtqx — COMMENTED** [review]
@ypan710 I've retested this PR and it seems that a new input element has been added. There are other alignment issues now with that new field added however the blue assign team button looks good on different screen sizes. 

Here it is on mobile:
Mobile view

And on Mobile S:
Screen Shot 2023-09-27 at 11 29 22 AM

Tablet:
Screen Shot 2023-09-27 at 11 30 09 AM

With large screen however, there is a little issue. The alignment is perfect on screen size up to 1170:

1170Width

But larger than that some misalignment occurs:

Alignment

**Comment by Shubhu0** [comment]
Hi @ypan710 The button looks good and all the previous changes seem to have been addressed. I noticed that the new element has some different changes in the alignment based on the different formats of devices. But apart from that the button looks good! Good job!

image


image

**Comment by ypan710** [comment]
> @ypan710 I've retested this PR and it seems that a new input element has been added. There are other alignment issues now with that new field added however the blue assign team button looks good on different screen sizes.
> 
> Here it is on mobile: Mobile view
> 
> And on Mobile S: Screen Shot 2023-09-27 at 11 29 22 AM
> 
> Tablet: Screen Shot 2023-09-27 at 11 30 09 AM
> 
> With large screen however, there is a little issue. The alignment is perfect on screen size up to 1170:
> 
> 1170Width
> 
> But larger than that some misalignment occurs:
> 
> Alignment

@cvtqx The new element "X-XXX" was the result of merging of the development branch into my branch.

**Comment by ypan710** [comment]
> Hi @ypan710 The button looks good and all the previous changes seem to have been addressed. I noticed that the new element has some different changes in the alignment based on the different formats of devices. But apart from that the button looks good! Good job!
> 
> image
> 
> image

@Shubhu0 The new element "X-XXX" was the result of merging of the development branch into my branch.

**Review by Alforoan — APPROVED** [review]
Everything looks good. Nice!


https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/118868270/f6bf4ff7-059b-4555-a6dd-11175a47e8d4

**Comment by one-community** [comment]
Thank you all, moving to final review.

**Review by cvtqx — APPROVED** [review]
The assign team button looks good across different screen sizes. great job!
Screen Shot 2023-12-19 at 3 01 08 PM
Screen Shot 2023-12-19 at 3 02 12 PM
Screen Shot 2023-12-19 at 3 01 57 PM

**Comment by one-community** [comment]
Thank you all, merging!

**Pull request merged**